File: beginner_tutorials/scripts/simple_action_server.py
#!/usr/bin/env python

import roslib; roslib.load_manifest('beginner_tutorials')
import rospy
import actionlib
from beginner_tutorials.msg import *

class DoDishesServer():
	def __init__(self):
		self.server = actionlib.SimpleActionServer('do_dishes', DoDishesAction, self.execute, False)
		self.server.start()

	def execute(self, goal):
		print "Requesting dishwasher %d"%(goal.dishwasher_id)
		result = self.server.get_default_result()
		result.total_dishes_cleaned = 100
		print "Returning dishes cleaned %d"%(result.total_dishes_cleaned)
		self.server.set_succeeded(result)

if __name__ == '__main__':
	rospy.init_node('do_dishes_server')
	server = DoDishesServer()
	rospy.spin()

File: beginner_tutorials/src/simple_action_server.cpp
#include <beginner_tutorials/DoDishesAction.h>
#include <actionlib/server/simple_action_server.h>

typedef actionlib::SimpleActionServer<beginner_tutorials::DoDishesAction> Server;

void execute(const beginner_tutorials::DoDishesGoalConstPtr& goal, Server* as)
{
    as->setSucceeded();
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "do_dishes_server");
    ros::NodeHandle n;
    Server server(n, "do_dishes", boost::bind(&execute, _1, &server), false);
    server.start();

    ros::spin();

    return 0;
}
